### DISCUSSION TITLE
How to test custom `ToSql` for `Sqlite`?

### DISCUSSION THREAD

**Discussion by agostbiro** [diesel-rs/diesel#3538] [Q&A] [answered]
Hi,

I'm looking for help to test my custom `ToSql` implementation for SQLite as I can't seem to get the serialized value out of the `diesel::serialize::Output` instance that is passed to the `to_sql` method. This is my code:

```rust
let value: SqliteBindValue = None::<String>.into();
let mut metadata = ();
let mut output = Output::<Sqlite>::new(value, &mut metadata);
let is_null =
    <TestText as ToSql<Text, Sqlite>>::to_sql(&text, &mut output).unwrap();
```

Any pointers appreciated!

**Comment by weiznich** [comment, marked as answer]
It's an explicit design decision to make `diesel::serialize::Output` opaque for anything outside of diesel. In addition it also contains nothing that you can easily match on to see whether serialization is successful. 
For diesel we test the (de-)serialization implementations by just using an actual database. Either by doing round trips (which checks both serialization and deserialization once) or by comparing to known values inside of the query and loading the result of the comparison as boolean.
See here for an example from diesel: https://github.com/diesel-rs/diesel/blob/25afffb6f0d1537a496ffbc1842102aba7936dce/diesel_tests/tests/types_roundtrip.rs#L18-L53

**Reply by agostbiro** [reply]
Thanks, this worked!
